### PR TITLE
Bump required Python 3.7 -> 3.8, drop TF1 support

### LINKED ISSUE #1326: Drop older Python support
Via #487, we are now supporting Python >=3.7.

Do we want to bump this and require Python >=3.8 or later? See also the Python feature list in #487.

Similarly, Python >=3.8 is now required for i6_core: https://github.com/rwth-i6/i6_core/pull/409, https://github.com/rwth-i6/i6_core/issues/408

Same discussion for Python >=3.8 for i6_models: https://github.com/rwth-i6/i6_models/pull/11

### PR DESCRIPTION
**Pull request by NeoLegends**
Closes #1326

Opening this to run CI and such.

### CODE REVIEW

**Comment by NeoLegends** [comment]
By dropping support for 3.7 we also drop support for TF1.

It is my understanding that some folks might still be using TF1.x for some of their experiments. However, these experiments are probably also using ancient RETURNN versions, so I believe it's ok to drop support.

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,8 @@ or any changes which could potentially break or change the behavior of existing
 This is intentionally kept short. For a full change log, just see the Git log.
 
 
+## 2023-12-13: Bump min Python version from 3.7 to 3.8 ([issue #1326](https://github.com/rwth-i6/returnn/issues/1326))
```
Comment (albertz) [inline CHANGELOG.md:9]: 2024

**Review by albertz — CHANGES_REQUESTED** [review]

**Comment by albertz** [comment]
Can you also add a line to our readme (or where else do you think we should put it)?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -174,23 +174,11 @@ jobs:
           - TEST=TFUtil
           - TEST=tools
         include:
-          - action: TEST=TFUtil
-            python-version: 3.7
-            tf-version: 1.15.3
```
Comment (albertz) [inline .github/workflows/main.yml:179]: Why did you remove this?

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -221,7 +209,7 @@ jobs:
 
         pip install --user pytest
 
-        if [[ "${{matrix.tf-version}}" == 2.[0123].* || "${{matrix.tf-version}}" == 1.* ]]; then
+        if [[ "${{matrix.tf-version}}" == 2.[0123].* ]]; then
```
Comment (albertz) [inline .github/workflows/main.yml:212]: Why?

File: CHANGELOG.md [hunk 4]
```diff
@@ -6,6 +6,10 @@ or any changes which could potentially break or change the behavior of existing
 This is intentionally kept short. For a full change log, just see the Git log.
 
 
+## 2024-12-13: Bump min Python version from 3.7 to 3.8 ([issue #1326](https://github.com/rwth-i6/returnn/issues/1326))
+
+This also drops support for TF 1.x.
```
Comment (albertz) [inline CHANGELOG.md:11]: Why?

**Comment by NeoLegends** [comment]
> Why?

TF1 never supported Python > 3.7. See the CI log. There are no pip packages available for TF1 for 3.8 and onwards.

**Comment by albertz** [comment]
Please check with @curufinwe if dropping TF1 is ok.

File: README.rst [hunk 5]
```diff
@@ -72,3 +72,10 @@ Questions can also be asked on
 
 .. image:: https://github.com/rwth-i6/returnn/workflows/CI/badge.svg
     :target: https://github.com/rwth-i6/returnn/actions
+
+Dependencies
+============
+
+pip dependencies are listed in ``requirements.txt`` and ``requirements-dev``, although some parts of the code may require additional dependencies (e.g. ``librosa``, ``resampy``) on-demand.
```
Comment (albertz) [inline README.rst:79]:
```suggestion
Pip dependencies are listed in ``requirements.txt`` and ``requirements-dev``,
although some parts of the code may require additional dependencies (e.g. ``librosa``, ``resampy``) on-demand.

For TensorFlow-based setups, RETURNN requires TF >=2.

For PyTorch-based setups, RETURNN requires Torch >=1.
```
(I actually don't really know our min required Torch version... But this here would be some good starting point.)

**Comment by albertz** [comment]
In the failing test, can you filter out `[IWE] tensorflow/` from stdout? I think that should fix the test.

**Comment by albertz** [comment]
Ah, in the failing test checking the stdout, the `SyntaxWarning`s are also new. Let's either fix the syntax warning or filter those out as well.

**Review by albertz — APPROVED** [review]

**Pull request merged**